docs/key-issues/multiple_architectures.md

```diff
@@ -0,0 +1,344 @@
+user tries to run an binary that doesn't contain an ARM64 slice, but *does*
+contain an x86-64 slice, the x86-64 slice will be converted at runtime into an
+ARM64 binary. Complications can occur when only *some* of the binary is being
+converted (e.g., if the binary being executed is fat, but a dynamic library
+isn't).
+
+To support the transition to Apple Silicon/M1 (ARM64), Python has introduced a
+`universal2` architecture target. This is effectively a "fat wheel" format; the
+`.dylib` files contained in the wheel are fat binaries containing both x86-64
+and ARM64 slices.
+
+
+??? question "What's the deal with `universal2` wheels?"
+
+    The `universal2` fat wheel format has generated quite a bit of discussion,
+    and isn't well-supported by either packaging tools (e.g., there is no way
+    to install a `universal2` wheel from PyPI if thin wheels are also present)
+    or package authors (most numerical, scientific and ML/AI package authors do
+    not provide them). There are some arguments for and against supporting the
+    format or even defaulting to it.
+
+    Arguments for:
+
+    - While users with a technical background are usually aware of the CPU
+      architecture of their machine, less technical end users are often
+      unaware of this detail (or of the significance of this detail). The
+      universal2 wheel format allows users to largely ignore this detail, as
+      all CPU architectures for the macOS platform are accomodated in a single
+      binary artefact.
+    - macOS has developed an ecosystem where end users expect that any macOS
+      binary will run on any macOS machine, regardless of CPU architecture.
+      As a result, when building macOS apps (`.dmg` downloadable installers
+      or similar formats, produced by tools such as
+      [py2app](https://py2app.readthedocs.io) or
+      [briefcase](https://beeware.org/project/projects/tools/briefcase/)),
+      the person building the project must be accommodate all possible CPU
+      architectures where the code *could* be executed.
+    - If binary wheels are only available in "thin" format, any issues with
+      merging those wheels into fat equivalents for distribution purposes are
+      deferred to the end user. This can be problematic as it may require
+      expert knowledge about the package being merged (such as optional
+      modules or header files that may not be present in both thin artefacts).
+      Universal2 artefacts captures this knowledge by requiring the project
+      maintainers to resolve any merging issues.
+
+    Arguments against:
+
+    - `universal2` wheels are never necessary for end users installing into
+      a locally installed Python environment exclusively for their own use,
+      which is the default experience most users have with Python.
+    - The tradeoff between download size and disk space usage vs. the upside
+      for say a .dmg installer is bad - for a typical PyData stack it takes
```

Comment:
"The tradeoff... is bad" is highly subjective language.

Reply:
Isn't this already implied in the "for" and "against" labels above the lists? I'd expect your "for" argument to say kinda the opposite. It's a tradeoff, so how you make it is inherently subjective. Happy to rephrase though, but not sure how. What would you suggest?

Reply:
"Using universal2 wheels requires larger downloads and more disk space" is undeniably true. "Using universal2 requires a tradeoff between download size/disk space and user experience" is true for at least a subset of users. "This tradeoff is bad" is a value judgement. As I've made clear a couple of times in this thread, I consider the tradeoff of disk space to be entirely acceptable, and not "bad" at all.

Reply:
:thinking:

Arguments for:
- the tradeoff is good because XXX

Arguments against:
- the tradeoff is bad because YYY

The current wording in this bullet point is not saying "the tradeoff is bad", it's saying "here is a persuasive argument for why I assert that it's bad". As such, I argue that it's not subjective language, it's a short persuasive essay.

Reply:
Okay, saying "using universal2 wheels requires larger downloads and more disk space" followed by the "- for a typical PyData stack" is fine with me here.